sdjournal/journal.go

```diff
@@ -357,7 +357,7 @@ func NewJournalFromDir(path string) (j *Journal, err error) {
 		return nil, err
 	}
 	defer func() {
-		if err == nil {
+		if err != nil {
```

Comment:
right. Oh, defer functions and implicit returns can be confusing. I'll fix this.